flake.nix

```diff
@@ -16,28 +16,32 @@
       url = "github:nix-community/naersk";
       inputs = {
         nixpkgs.follows = "nixpkgs";
-        flake-utils.follows = "utils";
       };
     };
-
-    mach-nix = {
-      url = "github:DavHau/mach-nix/3.5.0";
-      inputs.nixpkgs.follows = "nixpkgs";
-      inputs.flake-utils.follows = "utils";
-      inputs.pypi-deps-db.follows = "pypi-deps-db";
-    };
-
-    pypi-deps-db = {
-      url = "github:DavHau/mach-nix/3.5.0";
-    };
   };
 
-  outputs = { self, nixpkgs, naersk, rust-overlay, mach-nix, pypi-deps-db, utils }:
+  outputs = { self, nixpkgs, naersk, rust-overlay, utils }:
     utils.lib.eachDefaultSystem (system:
       let
         overlays = [ (import rust-overlay) ];
         pkgs = import nixpkgs {
           inherit system overlays;
+          config.packageOverrides = pkgs:
+            {
+              maturin = pkgs.rustPlatform.buildRustPackage rec {
+                pname = "maturin";
+                version = "0.14.7";
+                src = pkgs.fetchFromGitHub {
+                  owner = "PyO3";
+                  repo = "maturin";
+                  rev = "v0.14.7";
+                  hash = "sha256-PCE4SvUrS8ass8UPc+t5Ix126Q4tB2yCMU2kWuCfr5Q=";
+                };
+                cargoHash = "sha256-ODMOJOoyra29ZeaG0yKnjPRwcjh/20VsgOz+IGZbQ/s=";
+                nativeBuildInputs = [ pkgs.pkg-config ];
+                doCheck = false;
+              };
+            };
```

Comment:
Pending PR in nixpkgs to remove this: https://github.com/NixOS/nixpkgs/pull/207064